app/controllers/hq/sponsorships_controller.rb

```diff
@@ -0,0 +1,37 @@
+class Hq::SponsorshipsController < HqController
+
+  def inactivate
+    sponsorship = Sponsorship.find(params[:id])
+
+    @sponsorship_inactivator = InactivateSponsorship.new(sponsorship: sponsorship,
+                                                         end_date: params[:sponsorship][:end_date])
+    inactivate_sponsorship and redirect_to hq_sponsor_path(sponsorship.sponsor_id)
```

Comment:
I see your point.
`sponsorship.sponsor_id`was used instead `sponsorship.sponsor` to skip an extra sql query.
I guess it is not worth the sacrifice?